src/browser/browser.js

```diff
@@ -67,27 +72,46 @@ define((require, exports, module) => {
     'F5': reload
   });
 
-  const openTab = items => {
-    const item = open();
-    return select(append(items, item), x => x == item);
-  }
+
+  const addTab = item => items => append(items, item);
+
+  const openTab = (items) =>
+    append(items, open({isSelected: true,
+                        isActive: true}));
 
   // If closing viewer, replace it with a fresh one & select it.
   // This avoids code branching down the pipe that otherwise will
   // need to deal with 0 viewer & no active viewer case.
-  const closeTab = items =>
-    items.count() > 1 ? remove(items) : items.set(0, toggle(open()));
+  const close = p => items =>
+    items.count() > 1 ? remove(items, p) :
+    items.set(0, open({isSelected: true, isActive: true}));
+
+  const closeTab = item =>
+    close(x => x.get('id') == item.get('id'));
+
 
   const edit = edit => cursor => cursor.update(edit);
+
+  const onSelectNext = throttle(edit(selectNext), 200)
```

Comment:
> Just checking: I thought we had to remove throttle because it was causing state bug due to delay. Did you manage to fix issue in a way that makes it safe to include throttle?

I do believe issue was actually caused by an animation frame issue that I've explained in #82 if that turns out not to be a case we can remove throttle as well.